pkg/postgres/walker/walker.go

```diff
@@ -278,10 +278,12 @@ func handleStruct(ctx context, schema *Schema, original reflect.Type) {
 			continue
 		}
 		opts := getPostgresOptions(structField.Tag.Get("sql"), schema.Parent == nil, ctx.ignorePK, ctx.ignoreUnique, ctx.ignoreFKs)
-
 		if opts.Ignored {
 			continue
 		}
+		if opts.PrimaryKey && structField.Type.Kind() != reflect.String {
```

Comment:
All children have idx (int) which is a primary key

Reply:
These aren't in the walker